src/map/clif.c

```diff
@@ -11416,20 +11416,39 @@ void clif_parse_ItemIdentify(int fd,struct map_session_data *sd) __attribute__((
 ///     -1 = cancel
 void clif_parse_ItemIdentify(int fd,struct map_session_data *sd)
 {
-	short idx = RFIFOW(fd,2);
+	short idx = RFIFOW(fd, 2) - 2;
 
 	if (sd->menuskill_id != MC_IDENTIFY)
 		return;
-	if( idx == -1 ) {// cancel pressed
+	if( idx < 0 ) {// cancel pressed
```

Comment:
Just to make sure, what value does it send when pressing cancel?

Before this commit, cancel was when `RFIFOW(fd, 2) == -1`. After this commit, it is when `RFIFOW(fd, 2) == 1` (because you moved the -2 to line 11419)